inc/cobs.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+
+/*
+Consistent Overhead Byte Stuffing (COBS) removes a specific value from a
+list of values. The removed value can then, for example, be used as a
+frame marker.
+
+Refer to: https://en.wikipedia.org/wiki/Consistent_Overhead_Byte_Stuffing
+
+COBS considerations:
+
+1. Configurable marker byte value. The current implementation assumes
```

Comment:
An offset of 0 in marker byte 3 means the same as an offset of 1 in marker byte 0: it specifies that 0 data bytes will follow and then a marker byte should be inserted (except if we arrived at the marker byte/end of encoded packet).

`[3, 3, 3]` would be encoded as `[(0), {0}, {0}, {0}, 3]`